### ISSUE TITLE
hadoop command in shell scripts give Kcov error

### ISSUE THREAD

**Issue by bhaskarrana** [SimonKagstrom/kcov#186]
Hi Simon,

As hadoop command in shell script throwing error
$ kcov /tmp/out test.sh
O/P
kcov: error: ]} ${JAVA8_HOME_CANDIDATES[ is not an integer
kcov: error: ]}'
 is not an integer
kcov: error: HADOOP_COMMON_DIR=./
 is not an integer
kcov: error: HADOOP_COMMON_LIB_JARS_DIR=lib
 is not an integer
kcov: error: HADOOP_COMMON_LIB_NATIVE_DIR=lib/native
 is not an integer
kcov: error: HDFS_DIR=./
 is not an integer

Following is the script
 cat test.sh

#!/bin/bash
hadoop fs -ls

Regards
Bhaskar Rana

**Comment by SimonKagstrom** [comment]
I really have no clue as to what causes this. Is the hadoop command a shell script as well?

**Comment by bhaskarrana** [comment]
Hi Simon,
I will provide this with better example, as it has nothing to do with hadoop command.Probably will post it soon.
RegardsBhaskar Rana 

    On Tuesday, 21 February 2017 10:34 PM, Simon Kagstrom <notifications@github.com> wrote:
 

 I really have no clue as to what causes this. Is the hadoop command a shell script as well?—

**Comment by bhaskarrana** [comment]
Hi Simon,

Here is sample 

cat test
#!/usr/bin/env bash

BIGTOP_DEFAULTS_DIR=${BIGTOP_DEFAULTS_DIR-/etc/default}
[ -n "${BIGTOP_DEFAULTS_DIR}" -a -r ${BIGTOP_DEFAULTS_DIR}/bigtop-utils ] && . ${BIGTOP_DEFAULTS_DIR}/bigtop-utils

JAVA6_HOME_CANDIDATES=(
    '/usr/lib/j2sdk1.6-sun'
    '/usr/lib/jvm/java-6-sun'
    '/usr/lib/jvm/java-1.6.0-sun-1.6.0'
    '/usr/lib/jvm/j2sdk1.6-oracle'
    '/usr/lib/jvm/j2sdk1.6-oracle/jre'
    '/usr/java/jdk1.6'
    '/usr/java/jre1.6'
)


This file gives error

cat testError.sh

.  /home/abcaxis/A//abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/test

**Output**
kcov: error: 3BIGTOP_DEFAULTS_DIR=/etc/default

This file donot give error
cat testNoError.sh
.  /home/abcaxis/A/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/test

The difference is the directory nested level
**/home/abcaxis/A//abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/test
 /home/abcaxis/A/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/abcdef/test**

Regards
Bhaskar Rana

**Comment by SimonKagstrom** [comment]
Yes, I'm afraid this is a bash PS4 limitation, which will cut the string after some character limit. Again, it might be good to test the `--bash-method=DEBUG`, although I don't think that will fix it either.
